cmd/plakar/subcommands/agent/agent.go

```diff
@@ -294,7 +287,7 @@ func (cmd *Agent) ListenAndServe(ctx *appcontext.AppContext) error {
 			read := func(v interface{}) (interface{}, error) {
 				if err := decoder.Decode(v); err != nil {
 					if isDisconnectError(err) {
-						cancel()
+						clientContext.Close()
```

Comment:
the Close() is already deferred line 258

Reply:
yeah but we still need to forcefully shut down the connection.  calling close twice is fine, only the first will really cancel the context.

Reply:
because we are in a different routine here that's defferred below.  it's a bit of a mess.